src/prog_models/prognostics_model.py

```diff
@@ -1085,4 +1087,344 @@ def optimization_fcn(params):
 
         # Reset noise
         self.parameters['measurement_noise'] = m_noise
-        self.parameters['process_noise'] = p_noise                
+        self.parameters['process_noise'] = p_noise   
+
+    def generate_surrogate_dmd(self, load_functions, first_output = None, threshold_keys = None, **kwargs):
+        """
+        Generate a surrogate model to approximate the higher-fidelity model 
+
+        Parameters
+        ----------
+        load_functions : list of callable functions
+            Each index is a callable loading function of (t) -> z used to predict future loading (output) at a given time (t)
+
+        Keyword Arguments
+        -----------------
+        Includes all keyword arguments from simulate_to_threshold (except save_pts), and the following additional keywords: 
+
+        dt : Number or function, optional
+            Same as in simulate_to_threshold; for DMD, this value is the time step of the training data\n
+        save_freq : Number, optional
+            Same as in simulate_to_threshold; for DMD, this value is the time step with which the surrogate model is generated  \n
+        data_len: int, optional
+            Value between 0 and 1 that determines fraction of data resulting from simulate_to_threshold that is used to train DMD surrogate model
```

Comment:
Setting data_len (now trim_data) should limit each data trace to its respective percentage length, so each trace can still have different lengths, but it will be data_len% of the original length ... does that look right? 

In the end, I decided not to include the functionality to make all datasets the same length, because I figured the user could enforce this with the configuration of the training data, by setting the horizon to be the same for all training data. But, I didn't test that. I could add an example on it? Or we could add equal_lengths, that's not a bad idea. Open to your suggestions @teubert